rfcs/text/0004_application_service_mounting.md

```diff
@@ -13,16 +13,16 @@ first-class applications.
 import React from 'react';
 import ReactDOM from 'react-dom';
 
-import { MyApp } from './components';
-
 class MyPlugin {
   setup({ application }) {
     application.register({
       id: 'my-app',
       title: 'My Application',
-      mount(context, targetDomElement) {
+      async mount(context, targetDomElement) {
+        const { MyApp } = await import('./components');
```

Comment:
I don't think this is the ideal pattern in entirety, but I do think the RFC should show the ideal pattern.

The ideal pattern is that _all_ of the mounting logic of an application is deferred to an async import to limit the opportunity for plugins to explode their global plugin bundle size.

So the ideal pattern involves moving all of the react imports and ReactDOM.render call to a separate application bundle. Something like:

```js
class MyPlugin {
  setup({ application }) {
    application.register({
      id: 'my-app',
      title: 'My Application',
      async mount(context, targetDomElement) {
        const { renderApp } = await import('./application');
        return renderApp(context, targetDomElement);
      }
    });
  }
}
```

```js
import React from 'react';
import ReactDOM from 'react-dom';

export function renderApp(context, targetDomElement) {
  ReactDOM.render(
    <MyApp mountContext={context} deps={pluginStart} />,
    targetDomElement
  );
}
```